src/Storages/ObjectStorageQueue/StorageObjectStorageQueue.cpp

```diff
@@ -227,19 +230,55 @@ StorageObjectStorageQueue::StorageObjectStorageQueue(
     resolveSchemaAndFormat(columns, configuration->format, object_storage, configuration, format_settings, sample_path, context_);
     configuration->check(context_);
 
+    if ((*queue_settings_)[ObjectStorageQueueSetting::use_hive_partitioning])
+    {
+        HivePartitioningUtils::extractPartitionColumnsFromPathAndEnrichStorageColumns(
+            columns,
+            hive_partition_columns_to_read_from_file_path,
+            configuration->getRawPath().path,
+            true,
```

Comment:
> true

Do I get it right that one can not specify the schema of an S3Queue and it'll always be inferred from sample data?

Reply:
It's a valid case when S3Queue created before any data is written on storage. So creation must work without any sample data, just from `path` parameter.
But for consistency with S3 better to make this as a part of schema too, agree. I'll try to make this change.

Reply:
Add check that all hive columns are in schema